File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.py
import sys

input = sys.stdin.readline

n,m = map(int, input().split())
arr = [int(i) for i in input().split()]

dp = [arr[0]]
for i in range(1, n):
    dp.append(dp[-1] + arr[i])
    
for _ in range(m):
    a,b = map(int, input().split())
    result = dp[b - 1] - (dp[a - 2] if a - 2 >= 0 else 0)
    print(result)
    

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.js
const fs = require('fs');
const read = fs.readFileSync('dev/stdin').toString().trim();

let [nm, arr, ...abs] = read.split("\n");

let [n,m] = nm.split(" ").map(Number);
arr = arr.split(" ").map(Number);

let dp = [arr[0]];
for(let i = 1; i < n; i++ ){
    dp.push(dp.at(-1) + arr[i]);
}

for(let ab of abs) {
    let [a,b] = ab.split(" ").map(Number);
    let result = dp[b - 1] - (a - 2 >= 0 ? dp[a - 2] : 0)
    console.log(result);
}